server/data/data-jmap/src/main/java/org/apache/james/jmap/api/filtering/impl/DefineRulesCommandHandler.java

```diff
@@ -46,7 +46,7 @@ public DefineRulesCommandHandler(EventStore eventStore) {
         FilteringAggregateId aggregateId = new FilteringAggregateId(storeCommand.getUsername());
 
         return Mono.from(eventStore.getEventsOfAggregate(aggregateId))
-        .map(history -> FilteringAggregate.load(aggregateId, history).defineRules(storeCommand.getRules()));
+            .map(history -> FilteringAggregate.load(aggregateId, history).defineRules(storeCommand.getRules(), storeCommand.getIfInState()));
```

Comment:
Can we refactor to:

```suggestion
            .map(history -> FilteringAggregate.load(aggregateId, history).defineRules(storeCommand));
```

As the goal of an aggregate is to handle command?